app/models/blended_search.rb

```diff
@@ -44,40 +44,46 @@ def first_page?
   protected
 
   def handle_response(response)
-    if response
-      @total = response.total
-      post_processor = ResultsWithBodyAndDescriptionPostProcessor.new(response.results)
-      post_processor.post_process_results
-      @results = paginate(response.results)
-      @normalized_results = post_processor.normalized_results
-      @startrecord = ((@page - 1) * @per_page) + 1
-      @endrecord = @startrecord + @results.size - 1
-      assign_spelling_suggestion_if_eligible(response.suggestion.text) if response.suggestion.present?
-    end
+    return unless response
+
+    @total = response.total
+    post_processor = ResultsWithBodyAndDescriptionPostProcessor.new(response.results)
+    post_processor.post_process_results
+    @results = paginate(response.results)
+    process_data_for_redesign(post_processor)
+    @startrecord = ((@page - 1) * @per_page) + 1
+    @endrecord = @startrecord + @results.size - 1
+    assign_spelling_suggestion_if_eligible(response.suggestion.text) if response.suggestion.present?
   end
 
   def populate_additional_results
+    return unless first_page?
+
     @govbox_set = GovboxSet.new(query,
                                 affiliate,
                                 @options[:geoip_info],
-                                @highlight_options) if first_page?
+                                @highlight_options)
   end
 
   def log_serp_impressions
-    @modules << "LOVER" << "SPEL" unless self.spelling_suggestion.nil?
-    @modules |= (@govbox_set.modules - %w(NEWS)) if @govbox_set
+    @modules << 'LOVER' << 'SPEL' unless spelling_suggestion.nil?
+    @modules |= (@govbox_set.modules - %w[NEWS]) if @govbox_set
     @modules << modules_in_results if @total > 0
   end
 
   private
 
+  def process_data_for_redesign(post_processor)
```

Comment:
This new method (called in L53) is the only change in this file. The rest is Rubocop.